File: Python/LinkedList/src/Book.py

from Author import Author

class Book:
    def __init__(self, title = "", author = Author("",""), numberPages = 0):
        self.__title = title
        self.__author = author
        self.__numberPages = numberPages

    def getAuthor(self):
        return self.__author

    def getNumberPages(self):
        return self.__numberPages

    def getTitle(self):
        return self.__title

    def toString(self):
        return "Author: " + self.__author.getLastName() + "\n" + "Title: " + self.__title + "\n" + "Pages: " + str(self.__numberPages) + "\n\n"

File: JavaScript/LinkedList/src/Book.js

  function Book(_title, _author, _numberPages) {

    var self = this;
    self.title = _title;
    self.author = _author;
    self.numberPages = _numberPages;

    function getAuthor() {
      return self.author;
    }

    function getNumberPages() {
      return self.numberPages;
    }

    function getTitle() {
      return self.title;
    }

    function toString() {
      return "Author: " + self.author.getLastName() + "\n" + "Title: " + self.title + "\n" + "Pages: " + self.numberPages + "\n\n";
    }

    return {
      getAuthor: getAuthor,
      getNumberPages: getNumberPages,
      getTitle: getTitle,
      toString: toString
    }
  }

  module.exports = Book;
